### PR TITLE
[record-minmax] Add integer generator of random data

### PR DESCRIPTION
**Pull request by m-bronnikov**
This commit adds helper to generate integer random input data.

ONE-DCO-1.0-Signed-off-by: Maksim Bronnikov max120199@gmail.com

-------------------------

For: #8065
Draft: #8171

### CODE REVIEW

File: compiler/record-minmax/src/RecordMinMax.cpp [hunk 1]
```diff
@@ -388,28 +403,45 @@ void RecordMinMax::profileDataWithRandomInputs(const std::string &mode, float mi
...
+        }
+        case DataType::S64:
+        {
+          auto input_data = genRandomIntData<int64_t>(gen, num_elements);
+          _interpreter->writeInputTensor(input_node, input_data.data(),
+                                         input_data.size() * sizeof(int64_t));
+          break;
+        }
+        default:
+          throw std::runtime_error("Unsupported input type, yet!");
```
Comment (m-bronnikov) [inline compiler/record-minmax/src/RecordMinMax.cpp:444]: For reviewer: note this PR introduces throw of `runtime_error()` in case of unsupported input datatype and removes assert, which checks datatypes (see above).

File: compiler/record-minmax/src/RecordMinMax.cpp [hunk 2]
```diff
@@ -388,28 +403,45 @@ void RecordMinMax::profileDataWithRandomInputs(const std::string &mode, float mi
       if (num_elements == 0)
         throw std::runtime_error("Only support non-zero sized inputs");
 
-      // TODO Support more input data types
```
Comment (m-bronnikov) [inline compiler/record-minmax/src/RecordMinMax.cpp:391]: Hmm, I think better to leave this comment.

File: compiler/record-minmax/src/RecordMinMax.cpp [hunk 3]
```diff
@@ -388,28 +403,45 @@ void RecordMinMax::profileDataWithRandomInputs(const std::string &mode, float mi
...
-      assert(input_node->dtype() == loco::DataType::FLOAT32 ||
-             input_node->dtype() == loco::DataType::BOOL);
-
-      if (input_node->dtype() == DataType::FLOAT32)
+      switch (input_node->dtype())
```
Comment (seanshpark) [inline compiler/record-minmax/src/RecordMinMax.cpp:406]: this and below lines dones' seem to be related with the title/explanation.
please put these changes to another PR.

**Comment by seanshpark** [comment]
This PR title starts with `[luci-interpreter] ` where the file is in `record-minmax`
Please correct this.

File: compiler/record-minmax/src/RecordMinMax.cpp [hunk 4]
```diff
@@ -62,6 +62,21 @@ std::vector<uint8_t> genRandomBoolData(std::mt19937 &gen, uint32_t num_elements)
   return input_data;
 }
 
+template <typename T> std::vector<T> genRandomIntData(std::mt19937 &gen, uint32_t num_elements)
+{
+  std::uniform_int_distribution<T> dist(std::numeric_limits<T>::min(),
+                                        std::numeric_limits<T>::max());
```
Comment (jinevening) [inline compiler/record-minmax/src/RecordMinMax.cpp:68]: This code can generate extremely large values. Is it intended?

I don't know how the integer values will be used (maybe index?), but I think it would be better to restrict the range of generated values.

Comment (m-bronnikov) [reply compiler/record-minmax/src/RecordMinMax.cpp:68]: I thought about it, but I have not come up with the best way to do it (which range should be?), and for the OneHot operation, the input data (indices) can be any. So, I decided to use any integer values for input for this PR.

Do you have any suggestion to range of random input data? =)

Comment (m-bronnikov) [reply compiler/record-minmax/src/RecordMinMax.cpp:68]: @jinevening What do you think about [0, 127] range, which valid for any integer datatype?

Comment (jinevening) [reply compiler/record-minmax/src/RecordMinMax.cpp:68]: >  for the OneHot operation, the input data (indices) can be any.

For OneHot Op, I guess that the depth (range of indices) would be much smaller than the whole integer range. For example, for one-hot with depth = 1000, this random generator will make most of the output off-value.

How about using the range -100~100 (or -1000~1000)? It'll be possible to change the range if necessary. I think it would be better to take the range (min, max) as the arguments of this function.

Comment (m-bronnikov) [reply compiler/record-minmax/src/RecordMinMax.cpp:68]: > I think it would be better to take the range (min, max) as the arguments of this function.

I think this is a good approach.

Comment (jinevening) [reply compiler/record-minmax/src/RecordMinMax.cpp:68]: > What do you think about [0, 127] range, which valid for any integer datatype?

This looks also good to me :)

Comment (m-bronnikov) [reply compiler/record-minmax/src/RecordMinMax.cpp]: > > I think it would be better to take the range (min, max) as the arguments of this function.
> 
> I think this is a good approach.

I introduced 2 additional argument: min and max for generation helper. PTAL. =)

**Review by jinevening — APPROVED** [review]
LGTM

**Review by seanshpark — APPROVED** [review]
LGTM
